pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -335,12 +335,10 @@ func (og *operationGenerator) GenerateAttachVolumeFunc(
 
 		if attachErr != nil {
 			if derr, ok := attachErr.(*volerr.DanglingAttachError); ok {
-				addErr := actualStateOfWorld.MarkVolumeAsAttached(
+				addErr := actualStateOfWorld.MarkVolumeAsUncertain(
```

Comment:
nit: since the `else` statement below does the same thing `actualStateOfWorld.MarkVolumeAsUncertain(...)`.
Can you just have the `if`/`else` set what the node is, and then have common `actualStateOfWorld.MarkVolumeAsUncertain(...)` method. If not, at least update the `klog.Errorf("AttachVolume.MarkVolumeAsAttached...` below.

Reply:
fixed